react/src/hooks/useControllableState.ts

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import type { SetStateAction } from 'react';
+
+export interface Options<T> {
+  defaultValue?: T;
+  defaultValuePropName?: string;
+  valuePropName?: string;
+  trigger?: string;
+}
+
+export type Props = Record<string, any>;
+
+export interface StandardProps<T> {
+  value: T;
+  defaultValue?: T;
+  onChange: (val: T) => void;
+}
+/**
+ * Custom hook to  manage this kind of state. managed by itself or controlled by it's parent
+ * useControllableState is based on useControllableValue.
+ * However if the value is undefined, the component is treated as an uncontrolled component
+ */
+// params 설명 더 작성하기
```

Comment:
```suggestion
```

Reply:
I modified it ! https://github.com/lablup/backend.ai-webui/pull/2572/files/b2947efe9edf3c46776e99d6d549e6094ee7c484#diff-e98538d8ef63e2ebd590961cc47a0a8badea1d72e19f7cf0cbe5a3501e0a97bbR23